File: security/src/main/java/top/bettercode/summer/security/userdetails/LoginListener.java
package top.bettercode.summer.security.userdetails;

import javax.servlet.http.HttpServletRequest;
import top.bettercode.summer.security.token.ApiToken;

/**
 * @author Peter Wu
 */
public interface LoginListener {

  default void beforeLogin(HttpServletRequest request, String grantType, String scope) {
  }

  default void afterLogin(ApiToken apiToken) {
  }

}

File: security/src/main/kotlin/top/bettercode/summer/security/userdetails/LoginListener.kt
package top.bettercode.summer.security.userdetails

import jakarta.servlet.http.HttpServletRequest
import top.bettercode.summer.security.token.StoreToken

/**
 * @author Peter Wu
 */
interface LoginListener {

    fun beforeLogin(request: HttpServletRequest, grantType: String, clientId: String, scope: Set<String>) {
    }

    fun afterLogin(storeToken: StoreToken, request: HttpServletRequest) {}
}
